packages/web3torrent/src/contexts/torrent-context.ts

```diff
@@ -0,0 +1,23 @@
+import {download, upload, web3torrent} from '../clients/web3torrent-client';
+import {useEffect} from 'react';
+
+import React from 'react';
+import {getTorrentUI} from '../utils/torrent-status-checker';
+
+export const TorrentContext = React.createContext<ReturnType<typeof useTorrentContext>>(undefined);
+
+export function useTorrentContext({initializationContext}) {
+  const {initialize, isInitialized} = initializationContext;
```

Comment:
Question. Is the intention here to use a React context? Or just to use the object value passed in? I think that without the `useContext` hook, this function just uses the value of the object passid in.

Reply:
Originally I was hoping to use the context normally like `const {initialize, isInitialized} = useContext(InitializationContext)`. The problem is that since the `useTorrentContext` hook is being called in the context of setting a value for `TorrentContext.Provider` react hasn't wired up `InitializationContext` yet, so `useContext(InitializationContext)` returns `undefined`.

To work around this I just pass the actual context object in the props. It's slightly hacky but it got things working 😛